lib/pbench/server/unpack_tarballs.py

```diff
@@ -0,0 +1,135 @@
+import glob
+from logging import Logger
+import os
+from pathlib import Path
+import tempfile
+from typing import NamedTuple
+
+from pbench.server import PbenchServerConfig
+from pbench.server.filetree import FileTree
+from pbench.server.report import Report
+from pbench.server.utils import get_tarball_md5
+
+
+class Results(NamedTuple):
+    total: int
+    success: int
+
+
+class Unpack:
+    LINKSRC = "TO-UNPACK"
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        """Unpacks Tarball in the INCOMING Directory
+
+        Args:
+            config: PbenchServerConfig configuration object
+            logger: A Pbench python Logger
+        """
+        self.config = config
+        self.logger = logger
+
+    def unpack(self, lb_args: int, ub_args: int) -> Results:
+        """Scans for tarballs in the TO-UNPACK subdirectories of the
+        ARCHIVE directory and unpacks them using FileTree.unpack() function.
+
+        Args:
+            lb_args: Lowerbound size of Bucket
+            ub_args: Upperbound size of Bucket
+
+        Returns:
+            Results tuple containing the counts of Total and Successful tarballs.
+        """
+        tarlist = glob.iglob(self.config.ARCHIVE / "*" / self.LINKSRC / "*.tar.xz")
```

Comment:
So this is a list of symlinks. Does that make any difference e.g. in the calculation of the sizes or is the symlink resolved?

Reply:
By default, the `Path` reference is resolved to the linked-to file.